added solutions related to task-12

b-husein opened the pull request:
Hello, I have completed task-12. In this PR I have included all files and screenshots related to task solutions in separate directories like `01-packer`, `02-cloudformation`, `03-terraform`, `04-ansible` etc. There is also directory with some screenshots that I consider to be important.

sqlheisenberg commented on task-12/03-terraform/instances.tf:
```diff
@@ -0,0 +1,76 @@
+  }
+}
+
+resource "aws_instance" "web_server" {
```
You can consider using EC2 module 
https://github.com/terraform-aws-modules/terraform-aws-ec2-instance/tree/master

sqlheisenberg commented on task-12/04-ansible/database-servers/mysql-yum-repo.sh:
```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
+
+ sudo rpm --import https://repo.mysql.com/RPM-GPG-KEY-mysql-2022
```
This is done through packer and we have yum repo enabled

sqlheisenberg commented on task-12/04-ansible/database-servers/mysql-yum-repo.sh:
```diff
@@ -0,0 +1,8 @@
+ wget http://dev.mysql.com/get/mysql57-community-release-el7-8.noarch.rpm
+ sudo yum localinstall -y mysql57-community-release-el7-8.noarch.rpm
+ sudo yum install -y mysql-community-server
+ sudo systemctl start mysqld
```
you can use module instead of this inside your task
https://docs.ansible.com/ansible/latest/collections/ansible/builtin/dnf_module.html

sqlheisenberg commented on task-12/04-ansible/database-servers/db-server-playbook.yml:
```diff
@@ -0,0 +1,21 @@
+        group: root
+        mode: 0777
+
+    - name: Run sh script
```
Script is not needed

sqlheisenberg commented on task-12/04-ansible/inventory.ini:
```diff
@@ -0,0 +1,7 @@
+[webservers]
```
We should consider using aws_ec2 inventory plugin instead 
https://docs.ansible.com/ansible/latest/collections/amazon/aws/aws_ec2_inventory.html

sqlheisenberg replied:
example: aws_inventory.yml
```
plugin: aws_ec2
regions:
  - "eu-central-1"
keyed_groups:
  - prefix: ec2
    key: tags.Terraform
  - prefix: name
    key: tags.Name
filters:
  instance-state-name : running
compose:
  ansible_host: public_ip_address

```

sqlheisenberg commented on task-12/04-ansible/web-servers/web-servers-playbook.yml:
```diff
@@ -0,0 +1,47 @@
+  hosts: webservers
+  become: true
+  tasks:
+    - name: Install Nginx
```
Da li je ovo moglo da bude dio jednog taska umjesto 3

b-husein commented:
Hello, these are notes related to improvements that were suggested:

First, I have: 

* [x] created S3 bucket and secured state locking with DynamoDB
* [x] created EC2 instances using EC2 Instance Terraform module
* [x] wrote playbook that acts as aws_ec2 dynamic inventory in Ansible

and after that:

* [x] updated code to use AMI & Region variables in Packer.
* [x] updated script name to avoid error after running Packer file.
* [x] updated outbound rules to improve security in CloudFormation template.
* [x] updated CF template code to use AMI, VPC and Subnet parameters.
* [x] updated CF template code for end-user to create his own key-pair. 
* [x] updated Terraform file structure
* [x] updated tags
* [x] used `data` bock to retrieve IDs for the existing subnets.
* [x] updated TF code to use variables for AMI, VPC and subnet.
* [x] removed MySQL installation script in Ansible and wrote a playbook that performs installation of latest version of MySQL. 
* [x] Formatted all syntax (json & yaml) as suggested in review.

You can read more about changes and updates I made in `notes.md` file. All of the changes I made were tested prior to being pushed to GitHub.
